.github/workflows/workflow.yml

```diff
@@ -7,91 +7,160 @@ jobs:
+      - name: "5. Install main and dev dependencies"
         run: |
           pip install -q .
           pip install -q -r requirements/dev.txt
 
-      - name: Run tests
+      - name: "5. Run tests"
         run: |
           python -m unittest
 
   docs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-
-    - name: Set up Python 3.6
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.6
-
-    - name: Prepare machine
-      run: |
-        sudo apt-get install -y pandoc
-        python -m pip install --upgrade pip
-
-    - uses: actions/cache@v1
-      id: cache
-      with:
-        path: ~/.cache/pip
-        key: docs-${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt', 'requirements/dev.txt') }}
-        restore-keys: |
-          docs-${{ runner.os }}-pip-
-
-    - name: Install dependencies
-      run: |
-        pip install -q .
-        pip install -q -r requirements/dev.txt
-
-    - name: Copy examples
-      run: |
-        make --directory ./docs copy-examples
-
-    - name: Generate documentation
-      run: |
-        make --directory ./docs generate
-        
-    - name: Build documentation website
-      run: |
-        make --directory ./docs html
-
-    - uses: actions/upload-artifact@v1
-      with:
-        name: docs
-        path: docs/build/html
+      - name: "1. Clone repository"
+        uses: actions/checkout@v2
 
-  deploy-docs:
+      - name: "2. Set up Python 3.6"
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+
+      - name: "3. Update pip and install pandoc"
+        run: |
+          sudo apt-get install -y pandoc
+          python -m pip install --upgrade pip
+
+      - name: "4. Attach cache for pip"
+        uses: actions/cache@v1
+        id: cache
+        with:
+          path: ~/.cache/pip
+          key: docs-${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt', 'requirements/release.txt') }}
+          restore-keys: |
+            docs-${{ runner.os }}-pip-
+
+      - name: "5. Install main and release dependencies"
+        run: |
+          pip install -q .
+          pip install -q -r requirements/release.txt
+
+      - name: "6. Copy examples"
+        run: |
+          make --directory ./docs copy-examples
+
+      - name: "7. Generate API documentation"
+        run: |
+          make --directory ./docs generate
+
+      - name: "8. Build documentation"
+        run: |
+          make --directory ./docs html
+
+      - name: "9. Store documentation artifact"
+        uses: actions/upload-artifact@v1
+        with:
+          name: docs
+          path: docs/build/html
+
+  release:
     needs: [tests, docs]
     runs-on: ubuntu-latest
     if: contains(github.ref, 'master')
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/download-artifact@v1
+      - name: "1. Clone repository"
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
```

Comment:
what does it do? fetches only top-level dirs?

Reply:
# Number of commits to fetch. 0 indicates all history.
    # Default: 1
    fetch-depth: ''

We can do 1, so last commit